src/webgpu/api/operation/command_buffer/render/culling_tests.spec.ts

```diff
@@ -0,0 +1,160 @@
+  ): GPURenderPipeline {
+    // Draw two triangles with different winding orders:
+    // 1. The top-left one is counterclockwise (CCW)
+    // 2. The bottom-right one is clockwise (CW)
+    const vertexModule = this.makeShaderModule('vertex', {
+      glsl: `#version 450
+            const vec2 pos[6] = vec2[6](vec2(-1.0f,  1.0f),
+                                        vec2(-1.0f,  0.0f),
+                                        vec2( 0.0f,  1.0f),
+                                        vec2( 0.0f, -1.0f),
+                                        vec2( 1.0f,  0.0f),
+                                        vec2( 1.0f, -1.0f));
+            void main() {
+                gl_Position = vec4(pos[gl_VertexIndex], 0.0, 1.0);
+            }`,
+    });
+
+    // gl_FragCoord of pixel(x, y) in framebuffer coordinate is (x + 0.5, y + 0.5). And we use
+    // RGBA8 format for the back buffer. So (gl_FragCoord.xy - vec2(0.5)) / 255 in shader code
+    // will make the pixel's R and G channels exactly equal to the pixel's x and y coordinates.
+    const fragmentModule = this.makeShaderModule('fragment', {
+      glsl: `#version 450
+      layout(location = 0) out vec4 fragColor;
+      void main() {
+         fragColor = vec4((gl_FragCoord.xy - vec2(0.5)) / 255, 0.0, 1.0);
+      }`,
+    });
+
+    return this.device.createRenderPipeline({
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      primitiveTopology: 'triangle-list',
+      rasterizationState: { frontFace, cullMode },
+      colorStates: [
+        {
+          format,
+          alphaBlend: { dstFactor: 'zero', srcFactor: 'one', operation: 'add' },
+          colorBlend: { dstFactor: 'zero', srcFactor: 'one', operation: 'add' },
+          writeMask: GPUColorWrite.ALL,
+        },
+      ],
+      sampleCount: 1,
+    });
+  }
+
+  create2DTextureForTest(format: GPUTextureFormat, size: number) {
+    return this.device.createTexture({
+      size: { width: size, height: size, depth: 1 },
+      format,
+      mipLevelCount: 1,
+      sampleCount: 1,
+      dimension: '2d',
+      usage: GPUTextureUsage.OUTPUT_ATTACHMENT | GPUTextureUsage.COPY_SRC,
+    });
+  }
+
+  run(
+    frontFace: GPUFrontFace,
+    cullMode: GPUCullMode,
+    isCCWTriangleCulled: boolean,
+    isCWTriangleCulled: boolean,
+    format: GPUTextureFormat
+  ): void {
+    const size = 4;
+    const texture = this.create2DTextureForTest(format, size);
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: texture.createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 1.0, a: 1.0 },
+          storeOp: 'store',
+        },
+      ],
+    });
+    pass.setPipeline(this.createPipelineForTest(frontFace, cullMode, format));
+    pass.draw(6, 1, 0, 0);
+    pass.endPass();
+
+    const topLeftBuffer = this.device.createBuffer({
```

Comment:
Let's make a helper function (can be in gpu_test.ts) to expect the color of a single pixel in a texture.

It might be easy to implement this on top of the existing expectSingleColor, but I'm not sure whether it can check the contents of a specified subrectangle of a texture (seems like not right now...)

Eventually we're also going to need something to check the values of depth/stencil (which probably will be done by sampling the texture in a shader?) but this can wait a bit. (Leave comments for any TODOs we think of, though.)